src/braket/circuits/gate.py

```diff
@@ -210,6 +211,37 @@ def __eq__(self, other):
     def __repr__(self):
         return f"{self.name}('qubit_count': {self._qubit_count})"
 
+    def __hash__(self):
+        return hash((self.name, self.qubit_count))
+
+    @staticmethod
+    def _str_to_gate(class_name: str) -> Gate:
+        """
+        Returns the class of Gate corresponding to the `class_name`.
+
+        Args:
+            class_name (str): The name of the gate to convert.
+
+        Returns:
+            Gate: A Gate of type corresponding to `class_name`.
+        """
+        # Rx_12 does not exist in the Braket SDK, it is a gate between |1> and |2>
+        rigetti_gates_to_bdk_gates = {
+            "Rx_12": None,
+            "Cz": "CZ",
+            "Cphaseshift": "CPhaseShift",
+            "Xy": "XY",
+        }
+        if class_name in rigetti_gates_to_bdk_gates.keys():
+            class_name = rigetti_gates_to_bdk_gates[class_name]
```

Comment:
Why does the gate class need to hold a device specific map?

Reply:
Changed to not be a device specific map. Converting these gates will not be specific to a particular QPU